### PR TITLE
9 get articles

### PR DESCRIPTION
**Pull request by Dantren18**

### CODE REVIEW

**Comment by Dantren18** [comment]
Please note that my tests are failing here:

image

This is for a previous endpoint which I had working but when I changed branches it lost the progress. So when I merge I will combine the two.

File: __tests__/app.test.js [hunk 1]
```diff
@@ -3,6 +3,8 @@ const db = require("../db/connection");
 const seed = require("../db/seeds/seed");
 const data = require("../db/data/test-data/index");
 const request = require("supertest");
+const { expect } = require("@jest/globals");
```
Comment (popatre) [inline __tests__/app.test.js:6]: Not sure tou need this - it looks like an auto-import

File: __tests__/app.test.js [hunk 2]
```diff
@@ -20,119 +22,237 @@ describe("App", () => {
...
+        .patch("/api/articles/2")
+        .expect(422)
+        .then((res) => {
+          expect(res.text).toBe("Unprocessable Entity");
+        });
+    });
+    test("200 - request body includes other unrelated property", () => {
+      return request(app)
+        .patch("/api/articles/5")
+        .send({ inc_votes: 0, favouritePet: "dogs" })
```
Comment (popatre) [inline __tests__/app.test.js:204]: - [ ] TEST - patch to an valid article id, but the article doesnt exist e.g.  .patch("/api/articles/500000"
- [ ] TEST - patch to an invalid article id e.g.  .patch("/api/articles/sausages"

File: models/models.js [hunk 3]
```diff
@@ -1,16 +1,63 @@
...
+    !newVote.hasOwnProperty("inc_votes") ||
+    Number.isNaN(Number.parseInt(newVote.inc_votes))
+  ) {
+    return Promise.reject({ status: 422, message: "Unprocessable Entity" });
+  }
+
+  return db
+    .query(`SELECT * FROM articles WHERE article_id = $1`, [article_id])
+    .then((result) => {
+      return db.query(
```
Comment (popatre) [inline models/models.js:50]: If the article id is 900000, it will still get into the then with an empty array - what happens then?

Need to refactor this.

**Review by popatre — CHANGES_REQUESTED** [review]

File: __tests__/app.test.js (hunk 1 above)
Comment (Dantren18) [reply __tests__/app.test.js:6]: yep no idea when / how that for there! Deleted it

File: __tests__/app.test.js (hunk 2 above)
Comment (Dantren18) [reply __tests__/app.test.js:204]: added

File: models/models.js (hunk 3 above)
Comment (Dantren18) [reply models/models.js:50]: covered in above test

**Review by popatre — APPROVED** [review]
Much improved, well done!

**Pull request merged**